src/components/ContentWrapper.jsx

```diff
@@ -1,51 +1,80 @@
-import React from 'react'
-import { DOCTYPE_CONTACTS } from '../helpers/doctypes'
-import ContactsListDataLoader from './ContactsList/ContactsListDataLoader'
+import React, { useState } from 'react'
+import { PropTypes } from 'prop-types'
+
+import { useClient } from 'cozy-client'
+import log from 'cozy-logger'
+import { Content } from 'cozy-ui/transpiled/react/Layout'
+
 import Header from './Header'
 import Toolbar from './Toolbar'
-import { Content } from 'cozy-ui/transpiled/react/Layout'
-import { Query } from 'cozy-client'
-
-const ContentWrapper = () => (
-  <Query
-    query={client =>
-      client
-        .all(DOCTYPE_CONTACTS)
-        .include(['accounts'])
-        .where({
-          $or: [
-            {
-              trashed: {
-                $exists: false
-              }
-            },
-            {
-              trashed: false
-            }
-          ],
-          _id: {
-            $gt: null
-          }
-        })
-        .indexFields(['_id'])
-    }
-  >
-    {({ data: contacts, fetchStatus, hasMore, fetchMore }) => {
-      return (
-        <>
-          {contacts.length >= 1 && <Header right={<Toolbar />} />}
-          <Content>
-            <ContactsListDataLoader
-              contacts={contacts}
-              fetchStatus={fetchStatus}
-              hasMore={hasMore}
-              fetchMore={fetchMore}
-            />
-          </Content>
-        </>
-      )
-    }}
-  </Query>
-)
+import ContactsList from '../components/ContactsList/ContactsList.jsx'
+import SpinnerContact from './Components/Spinner'
+import {
+  useFetchContacts,
+  updateAndSortByFamilyNameGivenNameEmailCozyUrl
+} from '../helpers/contentWrapper'
+import {
+  contactsbyFamilyNameGivenNameEmailCozyUrl,
+  contactsWithoutIndexes
+} from '../helpers/queries'
+
+const ContentWrapper = ({ serviceToLaunch, isServiceLaunched }) => {
+  const client = useClient()
+  const [isContactsWithIndexesFinished, contactsWithIndexes] = useFetchContacts(
+    contactsbyFamilyNameGivenNameEmailCozyUrl
+  )
+  const [
+    isContactsWithNoIndexesFinished,
+    contactsWithNoIndexes
+  ] = useFetchContacts(contactsWithoutIndexes)
+
+  const [hasAlreadySortedContacts, setHasAlreadySortedContacts] = useState(
+    false
+  )
+  const [sortedContacts, setSortedContacts] = useState([])
+
+  const fetchesAreFinished =
+    isContactsWithIndexesFinished && isContactsWithNoIndexesFinished
+
+  // start keepIndexFullNameAndDisplayNameUpToDate service
+  // if never launched before
+  if (serviceToLaunch && isServiceLaunched === false) {
+    log(
+      'info',
+      `Executing keepIndexFullNameAndDisplayNameUpToDate service by Contacts app`
+    )
+    client.collection('io.cozy.triggers').launch(serviceToLaunch)
+  }
+
+  if (!fetchesAreFinished) {
+    return <SpinnerContact size="xxlarge" loadingType="fetching_contacts" />
+  }
+
+  if (fetchesAreFinished && !hasAlreadySortedContacts) {
+    const reworkedContacts = isServiceLaunched
```

Comment:
Reading this line I realise `isServiceFinished` or `hasServiceBeenLaunched` might be a more suitable name :)